src/utils/googleTagManager.ts

```diff
@@ -31,12 +30,17 @@ export const getAnonymizedLocation = ({ pathname, search, hash }: Location = his
   }
 
   // Anonymise transaction id
-  const txIdMatch = matchPath(pathname, { path: SAFE_ROUTES.TRANSACTIONS_SINGULAR })
+  const txIdMatch = matchPath<{ [TRANSACTION_ID_SLUG]: string }>(pathname, { path: SAFE_ROUTES.TRANSACTIONS_SINGULAR })
+
   if (txIdMatch) {
-    anonPathname = anonPathname.replace(txIdMatch.params[TRANSACTION_ID_SLUG], ANON_TX_ID)
+    const terms = new RegExp(/queue|history/)
+    const replacement = !terms.test(txIdMatch.params[TRANSACTION_ID_SLUG])
+      ? ANON_TX_ID
+      : txIdMatch.params[TRANSACTION_ID_SLUG]
+    anonPathname = anonPathname.replace(txIdMatch.params[TRANSACTION_ID_SLUG], replacement)
```

Comment:
This is a bit difficult to follow.

I suppose the intended meaning is something like this?

```
const txIdMatch = matchPath<{ [TRANSACTION_ID_SLUG]: string }>(pathname, { path: SAFE_ROUTES.TRANSACTIONS_SINGULAR })
const txId = txIdMatch.params[TRANSACTION_ID_SLUG]
if (txId !== 'queue' && txId !== 'history') {
  anonPathname = anonPathname.replace(txIdMatch.params[TRANSACTION_ID_SLUG], replacement)
}
```

However, I think a better fix would be to add a regex limit on the txId param in the router.

Something like
```
  TRANSACTIONS_SINGULAR: `${ADDRESSED_ROUTE}/transactions/:${TRANSACTION_ID_SLUG}(0x)`, // match only ids with 0x
```

Reply:
Agreed, looks cleaner to restrict the detail route but we have to add a few more keywords since we also support txIds that start with `multisig_` or `ethereum_`. Are there any more we would need to cover?

Reply:
If I rememer correctly, `matchPath`'s `path` property accepts and array. Maybe you could leverage that.

Reply:
I added another `matchPath` for the queue and history and removed the regex.